rts/motoko-rts/src/region.rs

```diff
@@ -948,6 +948,32 @@ pub(crate) unsafe fn region_load<M: Memory>(_mem: &mut M, r: Value, offset: u64,
     }
 }
 
+pub unsafe fn region_write_heap_shapshot<M: Memory>(mem: &mut M, r: Value) {
+    use crate::constants::WASM_PAGE_SIZE;
+    use core::arch::wasm32;
+
+    let heap_size: usize = wasm32::memory_size(0) * (WASM_PAGE_SIZE.as_u32() as usize);
+    if wasm32::memory_size(0) as u64 > region_size(mem, r) {
+        let new_pages = wasm32::memory_size(0) as u64 - region_size(mem, r);
+        region_grow(mem, r, new_pages);
+    };
+    if heap_size < isize::MAX as usize {
```

Comment:
Micro detail: Maybe a short comment could be helpful saying why this special case handling is needed (limitation of `from_raw_parts`).

Reply:
Sounds good.  I'll also add the same comment to the two other existing instances of this pattern (store/load Blob).